src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -6,13 +6,17 @@
 using System.Linq;
 using System.Security.Claims;
 using System.Text;
+using System.Text.RegularExpressions;
 using Microsoft.Extensions.Caching.Memory;
 using Microsoft.Extensions.Logging;
 
 namespace Microsoft.AspNetCore.Authentication.Negotiate;
 
-internal static class LdapAdapter
+internal static partial class LdapAdapter
 {
+    [GeneratedRegex(@"(?<!\\),")]
```

Comment:
Can a valid CN name end with a backlash? If so, we may need to also validate that the backlash is actually escaping the comma as opposed to be escaped by a previous backlash itself.

Reply:
Backslash can be escaped too (https://social.technet.microsoft.com/wiki/contents/articles/5312.active-directory-characters-to-escape.aspx). Consider this fact in regex.